roles/openshift_master/tasks/systemd_units.yml

```diff
@@ -38,32 +38,81 @@
   when: create_ha_unit_files | changed
 # end workaround for missing systemd unit files
 
+- name: Preserve Master API Proxy Config options
+  command: grep PROXY /etc/sysconfig/{{ openshift.common.service_type }}-master-api
+  register: master_api_proxy
+  when: openshift.master.ha is defined and openshift.master.ha | bool and openshift_master_cluster_method == "native"
+      and upgrading is defined and upgrading | bool
```

Comment:
I can't say I'm a fan of the upgrade variable here...  I don't want to hold this change up too much, since it is affecting users, but I'd like to make a point of cleaning this up at a later point...  maybe that can be rolled into @dgoodwin's upgrade changes?